src/arrayMethodJoin.js

```diff
@@ -5,7 +5,38 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    if (this.length === 0) {
+      return '';
+    }
+
+    if (separator === undefined) {
+      let str2 = this[0] + '';
+
+      for (let i = 1; i < this.length; i++) {
+        str2 += `,${this[i]}`;
+      }
+
+      return str2;
+    }
+
+    if (typeof separator === 'number') {
+      return this;
+    }
```

Comment:
do you really need this?

<img width="153" alt="image" src="https://user-images.githubusercontent.com/55802630/177111417-81b6694b-2721-4c0a-8fc0-0336e3bc716b.png">